R/react.R

```diff
@@ -9,20 +27,25 @@ Context <- R6Class(
     .invalidateCallbacks = list(),
     .flushCallbacks = list(),
     .domain = NULL,
+    .pid = NULL,
 
     initialize = function(domain, label='', type='other', prevId='') {
       id <<- .getReactiveEnvironment()$nextId()
       .label <<- label
       .domain <<- domain
+      .pid <<- processId()
       .graphCreateContext(id, label, type, prevId, domain)
     },
     run = function(func) {
       "Run the provided function under this context."
-      withReactiveDomain(.domain, {
-        env <- .getReactiveEnvironment()
-        .graphEnterContext(id)
-        on.exit(.graphExitContext(id), add = TRUE)
-        env$runWith(self, func)
+
+      promises::with_promise_domain(reactivePromiseDomain(), {
```

Comment:
We have nested `with_xx` functions in a bunch of places. Instead of doing that, we may want to do what withr does, which is provide a `local_xx` equivalent for each `with_xx` function. These functions would set the domain until the current function exits.

Reply:
Do you feel strongly enough about this to want to bring in the withr package as a dependency? It doesn't feel like that big of a deal to me to let these things nest a little... we're not asking our users to write code like this.

Reply:
No, I don't mean that we'd bring in withr. I just mean that we might be able to implement our own `local_x` functions that work similarly to the ones in withr, for reactive domains, promise domains, and restore contexts. Now that I look at the actual cases some more, I don't think it's worth the trouble.